docs/autoprocess.py

```diff
@@ -0,0 +1,412 @@
+            processes_in_file = read_yaml_file(yaml_file)
+            for process in processes_in_file:
+
+                # This section finds the line in file where the
+                # defintion of the process starts. (there are
+                # multiple process definition in some files).
+                i = 1
+                found = False
+                with open(yaml_file) as file_:
+                    for line in file_:
+                        if 'slug: ' + process['slug'] in line:
+                            found = True
+                            break
+                        i += 1
+                    if not found:
+                        # In case starting line is not found just
+                        # return first line of the file
+                        i = 1
+                startline = i
+
+                # Put together URL to process definition on GitHub:
+                cutted_yaml_path = yaml_file.split('/resolwe_bio/processes/')
+                github_url = github_base_url + cutted_yaml_path[1] + '#L' + str(startline)
+                chosen_process = [proc for proc in processes_in_file if proc['slug'] == process['slug']]
+                assert len(chosen_process) == 1
+
+                all_processes[github_url] = chosen_process[0]
+
+        return all_processes
+
+    def get_all_categories(self, process_list):
+        """
+        Get list of all categories in `process_list`.
+
+        For processes with undefined category field (yes, there are some!),
+        there is uncategorized category.
+
+        :param process_list: list of all processes
+        :type process_list: list
+
+        :rtype: list
+        """
+        categories = ['uncategorized']
+        for process in process_list:
+            if 'category' not in process or process['category'] in categories:
+                continue
+            else:
+                categories.append(process['category'])
+        return categories
+
+    def make_field(self, field_name, field_body):
+        """
+        Handy function for filling content into nodes.field
+
+        :param field_name: Field name of the field
+        :type field_name: string
+        :param field_name: Field body if the field
+        :type field_name: str or instance of docutils.nodes
+
+        :rtype: nodes.field
+        """
+        field = nodes.field()
+
+        name = nodes.field_name()
+        name += nodes.Text(field_name)
+
+        body = nodes.field_body()
+        paragraph = nodes.paragraph()
+        if isinstance(field_body, six.string_types):
+            # This is the case when field_body is just a string:
+            paragraph += nodes.Text(field_body)
+        else:
+            # This is the case when field_body is a complex node:
+            # useful when constructing nested field lists
+            paragraph += field_body
+
+        body += paragraph
+        field += name
+        field += body
+        return field
```

Comment:
Ok, then just define variables just before they are used. This way it is easier to figure it out where they come from.